panel/io/server.py

```diff
@@ -1015,57 +974,29 @@ def get_server(
     from ..config import config
     from .rest import REST_PROVIDERS
 
+    silence(EMPTY_LAYOUT, True)
     server_id = kwargs.pop('server_id', uuid.uuid4().hex)
-    kwargs['extra_patterns'] = extra_patterns = kwargs.get('extra_patterns', [])
-    if isinstance(panel, dict):
-        apps = {}
-        for slug, app in panel.items():
-            if slug.endswith('/') and not slug == '/':
-                raise ValueError(f"Invalid URL: trailing slash '/' used for {slug!r} not supported.")
-            if isinstance(title, dict):
-                try:
-                    title_ = title[slug]
-                except KeyError:
-                    raise KeyError(
-                        "Keys of the title dictionary and of the apps "
-                        f"dictionary must match. No {slug} key found in the "
-                        "title dictionary.") from None
-            else:
-                title_ = title
-            slug = slug if slug.startswith('/') else '/'+slug
-            if 'flask' in sys.modules:
-                from flask import Flask
-                if isinstance(app, Flask):
-                    wsgi = WSGIContainer(app)
-                    if slug == '/':
-                        raise ValueError('Flask apps must be served on a subpath.')
-                    if not slug.endswith('/'):
-                        slug += '/'
-                    extra_patterns.append(('^'+slug+'.*', ProxyFallbackHandler,
-                                           dict(fallback=wsgi, proxy=slug)))
-                    continue
-            if isinstance(app, pathlib.Path):
-                app = str(app) # enables serving apps from Paths
-            if (isinstance(app, str) and app.endswith(('.py', '.ipynb', '.md'))
-                and os.path.isfile(app)):
-                apps[slug] = app = build_single_handler_application(app)
-                app._admin = admin
-            elif isinstance(app, BkApplication):
-                apps[slug] = app
-            else:
-                handler = FunctionHandler(partial(_eval_panel, app, server_id, title_, location, admin))
-                apps[slug] = Application(handler, admin=admin)
-    else:
-        if isinstance(panel, pathlib.Path):
-            panel = str(panel) # enables serving apps from Paths
-        if isinstance(panel, BkApplication):
-            panel = {'/': app}
-        elif (isinstance(panel, str) and panel.endswith(('.py', '.ipynb', '.md'))
-            and os.path.isfile(panel)):
-            apps = {'/': build_single_handler_application(panel)}
-        else:
-            handler = FunctionHandler(partial(_eval_panel, panel, server_id, title, location, admin))
-            apps = {'/': Application(handler, admin=admin)}
+    kwargs['extra_patterns'] = extra_patterns = list(kwargs.get('extra_patterns', []))
+
+    def flask_handler(slug, app):
```

Comment:
Maybe expose this outside the method so people can hack and change it.

Reply:
It's a closure and I think it should be deprecated eventually.